l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -34,28 +34,50 @@ contract ExecutorFacet is Base, IExecutor {
     ) internal view returns (StoredBatchInfo memory) {
         require(_newBatch.batchNumber == _previousBatch.batchNumber + 1, "f"); // only commit next batch
 
+        uint8 pubdataSource = uint8(bytes1(_newBatch.pubdataCommitments[0]));
+        require(pubdataSource == uint8(PubdataSource.Calldata) || pubdataSource == uint8(PubdataSource.Blob), "us");
+
         // Check that batch contain all meta information for L2 logs.
         // Get the chained hash of priority transaction hashes.
-        (
-            uint256 expectedNumberOfLayer1Txs,
-            bytes32 expectedPriorityOperationsHash,
-            bytes32 previousBatchHash,
-            bytes32 stateDiffHash,
-            bytes32 l2LogsTreeRoot,
-            uint256 packedBatchAndL2BlockTimestamp
-        ) = _processL2Logs(_newBatch, _expectedSystemContractUpgradeTxHash);
-
-        require(_previousBatch.batchHash == previousBatchHash, "l");
+        LogProcessingOutput memory logOutput = _processL2Logs(
+            _newBatch,
+            _expectedSystemContractUpgradeTxHash
+        );
+
+        // TODO: Adapt to handle dynamic number of blobs
+        bytes32[] memory blobCommitments = new bytes32[](MAX_NUMBER_OF_BLOBS);
+        bytes32 blob1Hash;
+        bytes32 blob2Hash;
+        if (pubdataSource == uint8(PubdataSource.Blob)) {
+            // We want only want to include the actual blob linear hashes when we send pubdata via blobs.
+            // Otherwise we should be using bytes32(0)
+            blob1Hash = logOutput.blob1Hash;
+            blob2Hash = logOutput.blob2Hash;
+            bytes32[MAX_NUMBER_OF_BLOBS] memory blobHashes = [blob1Hash, blob2Hash];
```

Comment:
nit: you can just create this array once before the `if` statement (similar to the `blobCommitments` array)